### PR TITLE
#758: Remove "Atlas" from type names

### PR DESCRIPTION
**Pull request by bonkey**
- [x]  typealias CartCheckout = (cart: Cart?, checkout: Checkout?)
- [x] May be AtlasAPIWrapper or APIWrapper?
- [x] check translations
- [x] Try to unify AtlasUI and Atlas

### CODE REVIEW

File: AtlasUI/AtlasUI/Utils/AtlasUIClient.swift [hunk 1]
```diff
@@ -5,9 +5,11 @@
 import Foundation
 import AtlasSDK
 
+// TODO: Remove "Atlas" from name?
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/Utils/AtlasUIClient.swift]: actually I don't like the name as it is just a wrapper over the AtlasAPI to add a loader.
May be `AtlasAPIWrapper` or `APIWrapper`?

File: AtlasUI/AtlasUI/de.lproj/Localizable.strings [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-"AtlasCheckoutError.message.unclassified" = "Da ist etwas schief gelaufen.\nWir arbeiten an dem Problem.\nBitte komm später noch einmal vorbei.";
-"AtlasCheckoutError.title" = "Oops";
-"AtlasCheckoutError.title.missingAddress" = "Keine Adresse ausgwählt";
-"AtlasCheckoutError.title.priceChanged" = "Der Preis hat sich geändert";
-"AtlasConfigurationError.message.incorrectConfigServiceResponse" = "Server gab unbekannte Antwort zurück";
-"AtlasConfigurationError.message.missingClientId" = "Fehlender Client Identifier";
-"AtlasConfigurationError.message.missingInterfaceLanguage" = "Fehlende Anzeigensprache";
-"AtlasConfigurationError.message.missingSalesChannel" = "Sales Channel fehlt";
-"AtlasConfigurationError.title" = "Konfigurationsfehler";
+"APIError.message.checkoutFailed" = "unbekannter Checkout Fehler";
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/de.lproj/Localizable.strings]: playing with fire 🔥 
hope no problems will happen with POEditor

File: AtlasUI/AtlasUI/AddressFormActionHandler.swift [hunk 3]
```diff
@@ -32,7 +32,7 @@ extension AddressFormActionHandler {
                 return
         }
 
-        AtlasAPILoader.checkAddress(request) { result in
+        AtlasAPI.withLoader.checkAddress(request) { result in
```
Comment (bonkey) [inline AtlasUI/AtlasUI/AddressFormActionHandler.swift]: @hani-ibrahim Fancy, ah? :-)

I tried to play with such approach, but too hard for me, maybe you can get something smarter.

```swift
    func withLoader(_ call: ((@escaping APIResultCompletion<Customer>) -> Void),
                    completion: @escaping APIResultCompletion<Customer>) {
        AtlasUIViewController.displayLoader { hideLoader in
            call { result in
                hideLoader()
                completion(result)
            }
        }
    }
```

Comment (bonkey) [inline AtlasUI/AtlasUI/AddressFormActionHandler.swift]: I mean not for every function, but doing it with generics, like

```swift
    func withLoader<T>(_ call: ((@escaping T) -> Void),
                    completion: @escaping T) {
```

but different errors appeared.

**Comment by bonkey** [comment]
👍

File: AtlasUI/AtlasUI/GuestAddressActionHandler.swift [hunk 4]
```diff
@@ -5,14 +5,12 @@
...
 
-typealias GuestAddressActionHandlerCompletion = (_ address: EquatableAddress) -> Void
-
 class GuestAddressActionHandler {
 
     var addressCreationStrategy: AddressViewModelCreationStrategy?
     var emailAddress: String?
 
-    func createAddress(completion: @escaping GuestAddressActionHandlerCompletion) {
+    func createAddress(completion: @escaping Completion<EquatableAddress>) {
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/GuestAddressActionHandler.swift]: Nice 👍

File: AtlasUI/AtlasUI/Typealiases.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+typealias AddressCheckViewControllerCompletion = (AddressCheckResult) -> Void
+typealias AddressDataModelCreationStrategyCompletion = (AddressFormDataModel) -> Void
+
+typealias Email = String
+typealias AddressFormCompletion = (EquatableAddress, Email?) -> Void
+
+typealias AddressUpdatedHandler = (EquatableAddress) -> Void
+typealias AddressDeletedHandler = (EquatableAddress) -> Void
+typealias AddressSelectedHandler = (EquatableAddress) -> Void
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/Typealiases.swift]: why not using `Completion<xx>`?

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/Typealiases.swift (hunk 5 above)
Comment (bonkey) [inline AtlasUI/AtlasUI/Typealiases.swift]: They're used in few places, so decided to keep them. and my compiler was hanging when I tried using this:

```swift
 typealias AddressUpdatedHandler = Completion<EquatableAddress>
```

**Pull request merged**